alleycats-core/src/main/scala/alleycats/ReferentialEq.scala

```diff
@@ -0,0 +1,13 @@
+package alleycats
+
+import cats.Eq
+
+/**
+ * An `Eq[A]` that delegates to referential equality (`eq`).
+ * Note that it is not referentially transparent!
+ */
+object ReferentialEq {
+  def apply[A <: AnyRef]: Eq[A] = new Eq[A] {
+    def eqv(x: A, y: A) = x eq y
+  }
```

Comment:
can we add some tests? I think this actually passes the laws associated with Eq except `antiSymmetryEq`.

Maybe we could test the other three in an ad-hoc fashion?

Reply:
see: https://github.com/typelevel/cats/blob/main/kernel-laws/shared/src/main/scala/cats/kernel/laws/EqLaws.scala

Reply:
@johnynek good point, you are right! I just added it